lib/File/Which.pm

```diff
@@ -99,33 +99,65 @@ me.
 our @EXPORT    = 'which';
 our @EXPORT_OK = 'where';
 
-use constant IS_VMS => ($^O eq 'VMS');
-use constant IS_MAC => ($^O eq 'MacOS');
-use constant IS_WIN => ($^O eq 'MSWin32' or $^O eq 'dos' or $^O eq 'os2');
-use constant IS_DOS => IS_WIN();
-use constant IS_CYG => ($^O eq 'cygwin' || $^O eq 'msys');
-
-our $IMPLICIT_CURRENT_DIR = IS_WIN || IS_VMS || IS_MAC;
-
-# For Win32 systems, stores the extensions used for
-# executable files
-# For others, the empty string is used
-# because 'perl' . '' eq 'perl' => easier
-my @PATHEXT = ('');
-if ( IS_WIN ) {
-  # WinNT. PATHEXT might be set on Cygwin, but not used.
-  if ( $ENV{PATHEXT} ) {
-    push @PATHEXT, split /;/, $ENV{PATHEXT};
-  } else {
-    # Win9X or other: doesn't have PATHEXT, so needs hardcoded.
-    push @PATHEXT, qw{.com .exe .bat};
+sub _get_osname { @_ == 1 && ref $_[0] ? $_[0]->{osname} : $^O }
+
+sub IS_VMS { my $osname = &_get_osname; ($osname eq 'VMS'); }
+sub IS_MAC { my $osname = &_get_osname; ($osname eq 'MacOS'); }
+sub IS_WIN { my $osname = &_get_osname; ($osname eq 'MSWin32' or $osname eq 'dos' or $osname eq 'os2'); }
+sub IS_DOS { IS_WIN(@_); }
+sub IS_CYG { my $osname = &_get_osname; ($osname eq 'cygwin' || $osname eq 'msys'); }
```

Comment:
since these are private and no longer constants I think we should make these lowercase `is_cyg` etc.